internal/service/iam/openid_connect_provider.go

```diff
@@ -150,15 +154,18 @@ func resourceOpenIDConnectProviderUpdate(ctx context.Context, d *schema.Resource
 	conn := meta.(*conns.AWSClient).IAMClient(ctx)
 
 	if d.HasChange("thumbprint_list") {
-		input := &iam.UpdateOpenIDConnectProviderThumbprintInput{
-			OpenIDConnectProviderArn: aws.String(d.Id()),
-			ThumbprintList:           flex.ExpandStringValueList(d.Get("thumbprint_list").([]interface{})),
-		}
-
-		_, err := conn.UpdateOpenIDConnectProviderThumbprint(ctx, input)
-
-		if err != nil {
-			return sdkdiag.AppendErrorf(diags, "updating IAM OIDC Provider (%s) thumbprint: %s", d.Id(), err)
+		thumbprintList := d.Get("thumbprint_list").([]interface{})
+		if thumbprintList != nil {
+			input := &iam.UpdateOpenIDConnectProviderThumbprintInput{
+				OpenIDConnectProviderArn: aws.String(d.Id()),
+				ThumbprintList:           flex.ExpandStringValueList(thumbprintList),
+			}
+
+			_, err := conn.UpdateOpenIDConnectProviderThumbprint(ctx, input)
+
+			if err != nil {
+				return sdkdiag.AppendErrorf(diags, "updating IAM OIDC Provider (%s) thumbprint: %s", d.Id(), err)
+			}
```

Comment:
This is not correct, but I don't know how to fix it. 🤔 

If the IdP is implicitly trusted by AWS, it doesn't matter what the `thumbprint_list` is. So that's fine.

But if not, and the user unsets the attribute, the old value will now be kept both in AWS and in the state. There doesn't seem to be API call to trigger the automatic fetching of the thumbprint, so I guess the whole resource should be recreated in that case? I guess that should be done through the schema?